src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokenPsiReference.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider;
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReferenceBase;
+import com.intellij.util.ArrayUtil;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.features.navigation.LSPGotoDeclarationHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A PSI reference to a semantic token declaration that defers uses
+ * {@link LSPGotoDeclarationHandler#getGotoDeclarationTargets(PsiElement, int, Editor)} for deferred resolution.
+ */
+class LSPSemanticTokenPsiReference extends PsiReferenceBase<LSPSemanticTokenPsiElement> {
```

Comment:
So we have a `PsiElement` and an `IElementType`. We also need a `PsiReference` for those elements that are references. This is it.